### PR TITLE
BaseDraw: Use array to determine max width

### LINKED ISSUE #44: PHP 7.1: Uncaught Error: [] operator not supported for strings in (...)/vendor/szymach/c-pchart/src/BaseDraw.php on line 784
I'm getting the above mentioned fatal error with CpChart 3.0.0 and PHP 7.1.6.

The reason is that `BaseDraw` initialises a variable `$Width = ""`, but uses it like an `array`:
https://github.com/szymach/c-pchart/blob/08a2937e0b1f73f1105aa6fbd3f8873a1e43ed52/src/BaseDraw.php#L763-L784

### PR DESCRIPTION
**Pull request by caugner**
Fixes #44

### CODE REVIEW

File: src/BaseDraw.php [hunk 1]
```diff
@@ -760,7 +760,7 @@ public function getLegendSize(array $Format = [])
                     $vY = $vY + max($this->FontSize * count($Lines), $IconAreaHeight) + 5;
                 } elseif ($Mode == LEGEND_HORIZONTAL) {
                     $Lines = preg_split("/\n/", $Serie["Description"]);
-                    $Width = "";
+                    $Width = [];
```
Comment (caugner) [inline src/BaseDraw.php]: Alternatively we could replace this with `$MaxWidth = 0;`  and replace `$Width[] = $BoxArray[1]["X"];` with `$MaxWidth = max($MaxWidth, $BoxArray[1]["X"]);`

File: src/BaseDraw.php [hunk 2]
```diff
@@ -785,6 +785,7 @@ public function getLegendSize(array $Format = [])
                     }
 
                     $vX = max($Width) + $XStep;
+                    unset($Width);
```
Comment (szymach) [inline src/BaseDraw.php]: This is unnecessary, since it gets overwritten later on anyway.

File: src/BaseDraw.php (hunk 1 above)
Comment (szymach) [inline src/BaseDraw.php]: This is fine, it covers the original intent.

File: src/BaseDraw.php (hunk 2 above)
Comment (caugner) [inline src/BaseDraw.php]: Personally I prefer to be explicit here, because `$Width` below is an `int`, but I removed the line nevertheless.

**Pull request merged**